arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1207,6 +1207,7 @@ std::shared_ptr<LogicalView> ClusterInfo::getViewCurrent(
 std::vector<std::shared_ptr<LogicalView>> const ClusterInfo::getViews(
     DatabaseID const& databaseID) {
   std::vector<std::shared_ptr<LogicalView>> result;
+  std::set<TRI_voc_cid_t> dedup;
```

Comment:
Another approach is to mark each entry in `DatabaseViews` as primary or secondary and return only primary records. That will help us to avoid unnecessary heap allocations.